ddl_atomicity/ddl_atomicity_check_script_python2.py

```diff
@@ -0,0 +1,175 @@
+    # Fetch all user tables from pg_class for the database
+    pg_class_output = json.loads(
+        subprocess.check_output([
+            args.ysqlsh_path, 
+            "-h", args.ysql_host, 
+            "-d", dbname, 
+            "-t", 
+            "-c", "SELECT json_agg(row_to_json(t)) FROM (SELECT relname, oid, relfilenode FROM pg_class WHERE oid >= 16384) t;"
+        ]).decode().strip()
+    )
+    pg_class_oid_tableinfo_dict = {}
+    # Use relfilenode if it exists (as the table may be rewritten)
+    for table in pg_class_output:
+        if table['relfilenode'] != '0':
+            pg_class_oid_tableinfo_dict[table['relfilenode']] = table
+        else:
+            pg_class_oid_tableinfo_dict[table['oid']] = table
+
+    pg_attribute_output = json.loads(
+        subprocess.check_output([
+            args.ysqlsh_path, 
+            "-h", args.ysql_host, 
+            "-d", dbname, 
+            "-t", 
+            "-c", "SELECT json_agg(row_to_json(t)) FROM (SELECT attname, attrelid FROM pg_attribute WHERE attrelid >= 16384) t;"
+        ]).decode().strip()
+    )
+    pg_attribute_attrelid_attnames_dict = defaultdict(list)
+    for attribute in pg_attribute_output:
+        pg_attribute_attrelid_attnames_dict[attribute['attrelid']].append(attribute['attname'])
+
+    # Iterate through each table
+    for tablename, pg_oid, yb_pg_table_oid, tableid in tables:
+        print("\n\n")
+        # Check if the table exists in pg_class
+        if yb_pg_table_oid not in pg_class_oid_tableinfo_dict:
+            # Note: on versions older than 2024.1, the oid in this log will refer to the relfilenode
+            # for materialized views.
+            error_messages.append(
+                "❌ - Table {} with oid {} and uuid {} does not exist in database {} - ORPHANED TABLE NEEDS TO BE DROPPED".format(
+                    tablename, pg_oid, tableid, dbname
+                )
+            )
+            errorcount += 1
+            continue
+        
+        pg_class_entry = pg_class_oid_tableinfo_dict[yb_pg_table_oid]
+        # work-around for versions older than 2024.1, as master UI doesn't populate YSQL table oid
+        # on the UI correctly (it is populated with relfilenode oid instead).
+        pg_oid = pg_class_entry['oid']
+        
+        if tablename != pg_class_entry['relname']:
+            error_messages.append(
+                "❌ - Table {} with oid {} and uuid {} exists in {} but has a mismatched table name: table name in YSQL metadata is {} - TABLE NAME NEEDS TO BE FIXED".format(
+                    tablename, pg_oid, tableid, dbname, pg_class_entry['relname']
+                )
+            )
+            errorcount += 1
+        else:
+            print(
+                "✅ - Table {} with oid {} and uuid {} exists in database {} - NO ISSUE".format(
+                    tablename, pg_oid, tableid, dbname
+                )
+            )
+
+        # Get columns
+        table_schema_json = json.loads(
```

Comment:
What happens if the output from the `curl` comes back as something other than JSON?